lib/src/cli/install/install_command.dart

```diff
@@ -237,195 +125,82 @@ class InstallCommand extends Command<void> {
+  Future<String> _getRealmPackagePath() async {
     final packageConfig = await findPackageConfig(Directory.current);
     if (packageConfig == null) {
       throw Exception("Package configuration not found. "
-          "Run the 'dart run $packageName install` command from the root directory of your application");
+          "Run the 'dart run $_packageName install` command from the root directory of your application");
     }
 
-    final realmDartPackage = packageConfig.packages.where((p) => p.name == packageName).firstOrNull;
+    final realmDartPackage = packageConfig.packages.where((p) => p.name == _packageName).firstOrNull;
     if (realmDartPackage == null) {
-      throw Exception("$packageName package not found in dependencies. Add $packageName package to your pubspec.yaml");
+      throw Exception("$_packageName package not found in dependencies. Add $_packageName package to your pubspec.yaml");
     }
 
     if (realmDartPackage.root.scheme != 'file') {
-      throw Exception("$packageName package uri ${realmDartPackage.root} is not supported. Uri should start with file://");
+      throw Exception("$_packageName package uri ${realmDartPackage.root} is not supported. Uri should start with file://");
     }
 
     final realmPackagePath = path.join(realmDartPackage.root.toFilePath(), "pubspec.yaml");
     return realmPackagePath;
   }
 
-  Future<Pubspec> parseRealmPubspec(String path) async {
+  Future<Pubspec> _parseRealmPubspec(String path) async {
     try {
       var realmPubspecFile = await File(path).readAsString();
       final pubspec = Pubspec.parse(realmPubspecFile);
-      if (pubspec.name != packageName) {
-        throw Exception("Unexpected package name `${pubspec.name}` at $path. Realm install command expected package `$packageName`");
+      if (pubspec.name != _packageName) {
+        throw Exception("Unexpected package name `${pubspec.name}` at $path. Realm install command expected package `$_packageName`");
       }
 
       return pubspec;
     } on Exception catch (e) {
-      throw Exception("Error parsing package `$packageName` pubspect at $path. Error $e");
+      throw Exception("Error parsing package `$_packageName` pubspec at $path. Error $e");
     }
   }
 
   @override
   FutureOr<void>? run() async {
-    await runInternal();
-    print("Realm install comamnd finished.");
-  }
+    _options = parseOptionsResult(argResults!);
+    _validateOptions();
 
-  FutureOr<void>? runInternal() async {
-    options = parseOptionsResult(argResults!);
-    validateOptions();
+    final realmPackagePath = await _getRealmPackagePath();
+    final realmPubspec = await _parseRealmPubspec(realmPackagePath);
 
-    if (await skipInstall()) {
-      print("Realm install command started from within the realm-dart repo or one of the realm packages is a path dependency or has a path dependency override."
-          " Skipping install.");
+    if (realmPubspec.publishTo == 'none') {
```

Comment:
The previous implementation checked the package names in skipInstall as well. This one only checks publish to for skiping install. This makes this susceptible in the future the check to fail if someone forgets and commits in any realm package the `pubspec.yaml` file without `publish_to: none`. 
I suggest we keep the `skipInstall` implementation which also checked the package name

Reply:
Also I am not sure why are we allowing path dependencies for `realm` and `realm_dart` packages. What is the benefit here. Imo we should be checking for that as the previous version since this is most certainly a bug or we are the ones executing the install command. In such cases install should not happen

Reply:
The idea is to simplify using prerelease binaries here. Since we're not uploading those to pub.dev, the only other way for people to reference it is as a path dependency. I'd rather have us support that scenario, even if it's for internal use. It's a lot simpler to tell docs/product to download and unzip a package and then run install than to tell them where to unzip the binaries for each platform.

Reply:
I can bring back the package name check - not sure it brings a whole lot value, but it doesn't hurt either.